docs/tutorials/motion/plan-motion-with-arm-gripper.md

```diff
@@ -0,0 +1,431 @@
+
+You can pass additional information about the environment to various parts of the Viam system through a particular data structure, aptly named `WorldState`.
+The code samples below detail how to add geometry to the WorldState to indicate the presence of other objects in your robot's working environment.
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+<!-- TODO: Add Python -->
+```python {class="line-numbers linkable-line-numbers"}
+```
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+The `WorldState` is available through the `referenceframe` library, but additional geometry data must be added in a piecewise fashion.
+
+```go {class="line-numbers linkable-line-numbers"}
+// Add a table obstacle to a WorldState
+obstacles := make([]spatialmath.Geometry, 0)
+
+tableOrigin := spatialmath.NewPose(
+  r3.Vector{X: 0.0, Y: 0.0, Z: -10.0},
+  &spatialmath.OrientationVectorDegrees{OX: 0.0, OY: 0.0, OZ: 1.0, Theta: 0.0},
+)
+tableDims := r3.Vector{X: 2000.0, Y: 2000.0, Z: 20.0}
+tableObj, _ := spatialmath.NewBox(tableOrigin, tableDims, "table")
+obstacles = append(obstacles, tableObj)
+
+// Create a WorldState that has the GeometriesInFrame included
+obstaclesInFrame := referenceframe.NewGeometriesInFrame(referenceframe.World, obstacles)
+worldState := &referenceframe.WorldState{
+  Obstacles: []*referenceframe.GeometriesInFrame{obstaclesInFrame},
+}
+```
+{{% /tab %}}
+{{< /tabs >}}
+
+This example adds a "table" with the assumption that you may have mounted your robot arm to an elevated surface.
+The 2000 millimeter by 2000 millimeter dimensions ensure that a sufficiently large box is considered, regardless of the real physical footprint of your mounting surface.
+Feel free to change these dimensions, including thickness (the Z coordinate in the above code samples), to match your environment more closely.
+Additional obstacles can also be *appended* as desired.
+
+{{< alert title="Tip" color="note" >}}
+Within the App, the **Frame System** tab in the **Config** section of your robot gives you the ability to experiment with various geometry representations with better visual feedback.
+{{< /alert >}}
+
+<!-- TODO : Add a picture example of a geometry specified through the Frame System tab -->
+
+## Commanding an Arm to Move with the Motion Service
+
+Commanding motion with the Motion service has a more general feel than previous examples that were commanding motion for individual components.
+The Motion's `Move` method is what you will use when executing more general robotic motion.
+You can designate specific components to plan motion for by passing in the resource name (note the use of the arm resource in the code samples below).
+The `worldState` we constructed earlier is also passed in so that the Motion service can consider additional information when planning.
```

Comment:
```suggestion
The `worldState` we constructed earlier is also passed in so that the Motion Service can consider additional information when planning.
```